lib/ansible/modules/notification/rabbitmq_publish.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, John Imison <john+github@imison.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: rabbitmq_publish
+short_description: Publish a message to a RabbitMQ queue.
+version_added: "2.8"
+description:
+   - Publish a message on a RabbitMQ queue using a blocking connection.
+options:
+  url:
+    description:
+      - An URL connection string to connect to the RabbitMQ server.
+      - URL and host/port/user/pass/vhost are mutually exclusive, use either or but not both.
+  proto:
+    description:
+      - The protocol to use.  ampq or ampqs.
```

Comment:
`choices` must be used for `proto` parameter
```suggestion
      - The protocol to use.
  choices: [ampqs, ampq]
```

Reply:
Updated.  Thanks!